src/js/playlist/playlist.js

```diff
@@ -13,7 +13,8 @@ define([
     };
 
     /** Go through the playlist and choose a single playable type to play; remove sources of a different type **/
-    Playlist.filterPlaylist = function(playlist, providers, androidhls, configDrm, preload, feedid, withCredentials) {
+    Playlist.filterPlaylist = function(playlist, providers, androidhls, configDrm,
+                                       preload, feedid, withCredentials, hlsjsdefault) {
```

Comment:
Replace `androidhls, configDrm, preload, feedid, withCredentials, hlsjsdefault` with `model`.

`_formatSources` should take `item, model`